lib/http.ex

```diff
@@ -112,8 +112,8 @@ defmodule Braintree.HTTP do
   @doc false
   @spec build_url(binary, Keyword.t) :: binary
   def build_url(path, opts) do
-    environment = Keyword.get(opts, :environment, Braintree.get_env(:environment, :sandbox))
-    merchant_id = Keyword.get(opts, :merchant_id, Braintree.get_env(:merchant_id))
+    environment = opts[:environment] || Braintree.get_env(:environment, :sandbox)
```

Comment:
Seems like a perfect case for `Keyword.get_lazy/3` https://hexdocs.pm/elixir/Keyword.html#get_lazy/3